.changeset/ninety-forks-poke.md

```diff
@@ -0,0 +1,5 @@
+---
+"ts-essentials": patch
+---
+
+Fix `OptionalKeys<Type>`, it now works as expected with primitives and arrays
```

Comment:
I suggest being more specific on what was changed:

```suggestion
`OptionalKeys<Type>` returns `never` for primitives and returns only optional indices for arrays and tuples
```

Reply:
Agreed!
Updated the PR.